### PR TITLE
Add ActorMove and ActorSetPos lines

### PR DESCRIPTION
**Pull request by valarnin**
Ready for initial review. We should do some sort of rate limiting on `ActorMove`, as it can get very, very spammy during e.g. large FATEs. I'm open to thoughts here, my initial instinct was to limit to 1 line per actor per 500ms. Maybe only rate limit players.

`ActorSetPos` doesn't need any rate limiting, it's pretty rare.

Here's some example lines from doing an unsynced Ifrit (Hard):

```
270|2024-02-26T16:00:17.2200000-05:00|400120DD|-1.5882|0000|0050|14.6489|0.0000|0.0000|d1ddc8ce9596e013
270|2024-02-26T16:00:17.5340000-05:00|400120DD|-1.5962|0000|0050|13.6418|0.0000|-0.0305|935b038d8e441a4e
270|2024-02-26T16:00:21.1960000-05:00|400120DD|-1.2989|0000|0050|13.3061|0.0000|0.0916|a5ebe3da4c3244ce
270|2024-02-26T16:00:21.7340000-05:00|400120DD|-1.1570|0000|0050|13.3061|0.0000|0.0916|6fca70cfc14460dc
270|2024-02-26T16:00:22.0490000-05:00|400120DD|-0.8149|0000|0050|13.3061|0.0000|0.0916|f2895265899c114d
270|2024-02-26T16:00:22.3620000-05:00|400120DD|-0.3392|0000|0050|13.3061|0.0000|0.0916|0dffe177c0a46e8f
270|2024-02-26T16:00:22.6770000-05:00|400120DD|0.2639|0000|0050|13.3061|0.0000|0.0916|9e73801447836eb8
270|2024-02-26T16:00:22.9890000-05:00|400120DD|0.9809|0000|0050|13.3061|0.0000|0.0916|0377f9e74808526a
270|2024-02-26T16:00:23.3040000-05:00|400120DD|1.7533|0000|0050|13.3061|0.0000|0.0916|c563be4c53716c35
270|2024-02-26T16:00:27.8670000-05:00|400120DD|1.8108|0000|0050|13.3061|0.0000|0.0916|e6661a58015a6613
270|2024-02-26T16:00:37.9920000-05:00|400120DD|1.9228|0000|0050|13.3061|0.0000|0.0916|a653fa03b5829893
270|2024-02-26T16:00:38.7470000-05:00|400120DD|1.7821|0000|0050|13.3061|0.0000|0.0916|acfc6fa656313a96
271|2024-02-26T16:00:44.1860000-05:00|400120DD|-0.0001|00|00|0.0000|0.0000|0.0000|38b8bf79cc582848
270|2024-02-26T16:00:51.3290000-05:00|400120DD|0.5654|0000|0050|0.0000|0.0000|0.0000|625607a0bfb7f8e8
270|2024-02-26T16:00:51.6410000-05:00|400120DD|1.6439|0000|0050|0.0000|0.0000|0.0000|6597121488ccc350
270|2024-02-26T16:00:52.4020000-05:00|400120DD|1.6439|0000|0050|0.7019|0.0000|-0.0610|a2ed38d11c2f2322
270|2024-02-26T16:00:52.7140000-05:00|400120DD|1.6439|0000|0050|0.7935|0.0000|-0.0610|70f4efe3627bece0
270|2024-02-26T16:01:00.3900000-05:00|400120DD|1.6424|0000|0050|2.2279|0.0000|-0.1526|38299ce63410a20d
270|2024-02-26T16:01:00.7030000-05:00|400120DD|1.6424|0000|0050|4.6999|0.0000|-0.3357|ca3606edb8fb0b4a
270|2024-02-26T16:01:01.0160000-05:00|400120DD|1.6424|0000|0050|7.2024|0.0305|-0.5188|39125dcde1d1ae34
270|2024-02-26T16:01:01.3290000-05:00|400120DD|1.6424|0000|0050|9.7049|0.0305|-0.7019|26be0571a20e7a96
270|2024-02-26T16:01:01.6410000-05:00|400120DD|1.6424|0000|0050|10.7730|0.0305|-0.7630|d9c8d32d65c4caca
270|2024-02-26T16:01:02.3060000-05:00|400120D9|-0.3687|0002|0014|36.0118|0.0000|0.0000|21f52c98304d46b6
270|2024-02-26T16:01:02.6180000-05:00|400120D9|-1.6446|0002|0014|36.0118|0.0000|0.0000|eb74fc8f11cdf3ed
270|2024-02-26T16:01:04.3130000-05:00|400120D8|-0.5613|0002|0014|36.0118|0.0000|0.0000|c2b5b533dbfbcd0f
270|2024-02-26T16:01:04.6270000-05:00|400120D8|-1.6446|0002|0014|36.0118|0.0000|0.0000|c97d905c0720fef9
270|2024-02-26T16:01:06.3260000-05:00|400120D7|-0.5613|0002|0014|36.0118|0.0000|0.0000|c58538612be2b7bd
270|2024-02-26T16:01:06.6380000-05:00|400120D7|-1.6446|0002|0014|36.0118|0.0000|0.0000|7fa68b5691e97576
271|2024-02-26T16:01:11.1070000-05:00|400120DC|-0.0001|00|00|0.0000|-21.0000|0.0000|0f7302b94d2d2a0b
271|2024-02-26T16:01:11.1070000-05:00|400120DB|-0.7855|00|00|14.8500|-14.8500|0.0000|00e1f19f8218c657
271|2024-02-26T16:01:11.1070000-05:00|400120DA|-1.5705|00|00|21.0000|0.0000|0.0000|7050f3bcc9cd2e5c
270|2024-02-26T16:01:21.5870000-05:00|400120DD|1.6871|0000|0050|10.7730|0.0305|-0.7630|f7ef7ad84b077ce3
270|2024-02-26T16:01:21.8990000-05:00|400120DD|-2.3477|0000|0050|10.7730|0.0305|-0.7630|20e98f4198c6ea98
270|2024-02-26T16:01:22.2110000-05:00|400120DD|-1.5688|0000|0050|10.7730|0.0305|-0.7630|2499e30f13fb73f2
270|2024-02-26T16:01:22.5240000-05:00|400120DD|-1.5425|0000|0050|10.7730|0.0305|-0.7630|0675c211cab48c10
271|2024-02-26T16:01:23.1910000-05:00|400120DD|-1.4631|00|00|11.4747|-1.2412|0.0456|fd4cef80fe3d3771
```

Of particular note are these three lines, which correspond to the Ifrit clones spawning at the edge of the arena for dashes:

```
271|2024-02-26T16:01:11.1070000-05:00|400120DC|-0.0001|00|00|0.0000|-21.0000|0.0000|0f7302b94d2d2a0b
271|2024-02-26T16:01:11.1070000-05:00|400120DB|-0.7855|00|00|14.8500|-14.8500|0.0000|00e1f19f8218c657
271|2024-02-26T16:01:11.1070000-05:00|400120DA|-1.5705|00|00|21.0000|0.0000|0.0000|7050f3bcc9cd2e5c
```

### CODE REVIEW

**Comment by xpdota** [comment]
I think I had maybe commented in a previous discussion but I'm not sure.

This doesn't matter that much from my end, since I can just consume the positions from these lines like any other. But for everything else, I'd imagine that it would be better to work these into LineCombatant. LineCombatant already has rate limiting logic, and already conveys the same data. IMO if you have a source of data (in this case, ActorMove) that is more responsive (due to being driven by network data rather than memory), then that data source should supersede the less-responsive data (combatant memory reading) if available. The way I'd probably process these is the same way I do HP - HP data from 37/38/39-lines supersedes HP from any other source, since those are the packets that result in an HP change. Every other source of HP (other log lines, combatants memory) has potential issues of staleness.

In your testing, have you run into any scenarios where the memory data changes, but there isn't a corresponding ActorMove/ActorSetPos?

**Comment by wexxlee** [comment]
Still formulating my thinking on this, but initial reactions:

* Assuming `ActorSetPos` is fairly rare, I'm in favor of having it as a separate log line, as this PR does.  Ifrit dashes are a good example, and I suspect there are other mechanics that use ActorSetPos (trios in ucob?) where having that data available as a distinct, easily-regexable log line will be very useful for trigger writing.  I think it's just cleaner from the "consumer" standpoint.

* I am concerned about the spamminess of `ActorMove`.  xp's suggestion of combining with LineCombatant does have appeal, as I'm not sure of the benefit of having two different sets of position data from two different sources.  If `ActorMove` were going to be separate, I'd lean toward some additional constraints, like checking in-combat state (hard to imagine usefulness out of combat, and I shudder to think of the disk space massacre from Skyrise Celebrations 2.0), and a rate limit on all player combatants.

I'm planning to grab a build and do a little testing in some different content today and tomorrow, just to get some firmer data on the increased logging before those opinions set too firmly though.

**Comment by valarnin** [comment]
It would be fairly easy to wire `ActorMove` up to the existing `CombatantMemory` lines, but there are some `ActorMove` instances that we need immediately. The first that comes to mind is Titan jumps in UWU. The turn is an `ActorMove`, and the `ActorSetPos` doesn't fire until the landing animation starts and damage is already locked in.

**Comment by xpdota** [comment]
Got another question: is ActorMove something that the client is supposed to interpolate, in order to make actors move/turn in a smooth manner rather than jerky? If that's the case, I guess there is technically a small difference between the two, and that most triggers would actually prefer the "predicted" position rather than the current in-memory position.

> but there are some ActorMove instances that we need immediately

Couldn't CombatantMemory just have the same logic? What I'm thinking is that CombatantMemory could just do an immediate flush off that actor's data if whatever set of conditions is met (not a player && moved more than a certain amount?)

**Comment by valarnin** [comment]
> Got another question: is ActorMove something that the client is supposed to interpolate, in order to make actors move/turn in a smooth manner rather than jerky? If that's the case, I guess there is technically a small difference between the two, and that most triggers would actually prefer the "predicted" position rather than the current in-memory position.

Yes, the client interpolates between current position and the new position in `ActorMove`. The first `unknown` byte is some kind of flags or properties that determine how that interpolation should be performed, but the decompiled code around it was a giant mess so I couldn't make heads or tails of if that was the only thing contained in that field or if other data was included as well.

> Couldn't CombatantMemory just have the same logic? What I'm thinking is that CombatantMemory could just do an immediate flush off that actor's data if whatever set of conditions is met (not a player && moved more than a certain amount?)

What is "a certain amount"? If UWU Titan is already facing straight north, and he jumps north, there'll be an `ActorMove` line with the same exact x/y/z/heading as his current position, at which point the "distance" moved is `0` no matter how it's measured, but that's still an important line for writing a trigger for the mechanic.

**Comment by xpdota** [comment]
> What is "a certain amount"? If UWU Titan is already facing straight north, and he jumps north, there'll be an ActorMove line with the same exact x/y/z/heading as his current position, at which point the "distance" moved is 0 no matter how it's measured, but that's still an important line for writing a trigger for the mechanic.

If I'm understanding correctly, the issue you're talking about is something like:
1. Right before Titan jumps, tank moves a bit so titan is facing roughly north
2. Titan goes untargetable and an ActorMove packet comes in telling Titan to face exactly north but in the same position

I think this is an issue with both approaches. If you throttle or threshold the updates, you risk missing the second update due to it either being spaced too closely to the first update, or due to not reaching the "how much did they move/turn" threshold. However, I think it's still a little less problematic with the CombatantMemory approach, since the update will still *eventually* come, once the throttle timer has elapsed. It's theoretically possible to do that with log lines, but that's unexplored territory. Either way, a trigger would have to wait for Titan to become untargetable and wait a bit of time (realistically the same as how it works currently). 

It could have logic to force a flush when certain things happen (targetable/untargetable, cast starts, etc), but that probably doesn't need to happen for an initial release.

**Comment by valarnin** [comment]
> How would that `ActorMove` line be found among other unrelated `ActorMove` lines (normal dragging or turning of Titan)?

`Trigger A` has a condition of `data.watchingForTitanJump === true`. While active, it stores the most recent `ActorMove` line for the main `Titan` combatant (which can be detected at the start of the pull).
`Trigger B` toggles `data.watchingForTitanJump` to `true` when appropriate (maybe based on `timelineTriggers` or maybe based on ability use).
`Trigger C` watches for Titan to go untargetable, and if there's a stored `ActorMove` position, alerts to the direction, then disables `data.watchingForTitanJump` and clears the stored position.

Probably not the most clean way, maybe we can detect the jump animation somehow (maybe that's one of the `unknown` fields? more research required). But it's certainly possible.

Right now there's a note at the top of the UWU triggers file explaining that `getCombatants` is inconsistent due to timing issues due to the small window.

---



> If I'm understanding correctly, the issue you're talking about is something like:
> 
>     1. Right before Titan jumps, tank moves a bit so titan is facing roughly north
> 
>     2. Titan goes untargetable and an ActorMove packet comes in telling Titan to face exactly north but in the same position
> 

This issue is even easier to explain.

1. Phase into Titan
2. Tank is already straight north. Titan doesn't move or turn to face tank, since they're already in position
3. Tank never gets targeted for weight of the land, so they never have to dodge weight (or they reposition perfectly straight north)
4. Titan jumps north after weights

Position and rotation are identical in this case for the whole segment of the phase for the boss.

> I think this is an issue with both approaches. If you throttle or threshold the updates, you risk missing the second update due to it either being spaced too closely to the first update, or due to not reaching the "how much did they move/turn" threshold. However, I think it's still a little less problematic with the CombatantMemory approach, since the update will still _eventually_ come, once the throttle timer has elapsed. It's theoretically possible to do that with log lines, but that's unexplored territory. Either way, a trigger would have to wait for Titan to become untargetable and wait a bit of time (realistically the same as how it works currently).
> 
> It could have logic to force a flush when certain things happen (targetable/untargetable, cast starts, etc), but that probably doesn't need to happen for an initial release.

The initial thought was to only throttle for player lines. Enemy lines are always going to be more rare/sparse.

We could pipe the position data over to the CombatantMemory class so it has the info, or just have it stop watching for updates to position and heading since we'd be getting that data directly instead.

You asked before if position was ever sent in something else, the only other thing I know of is the `PlayerSpawn`/`ActorSpawn` packets, which get covered by the initial `03` lines that `FFXIV_ACT_Plugin` already emits, though that information is currently read from memory and not the packets themselves. I thought I included this info in a previous reply here but must have not.

**Comment by wexxlee** [comment]
> The initial thought was to only throttle for player lines. Enemy lines are always going to be more rare/sparse.
> 
> We could pipe the position data over to the CombatantMemory class so it has the info, or just have it stop watching for updates to position and heading since we'd be getting that data directly instead.

Still hoping to collect some data on volume (my plans were stymied by a keyboard short), but in the interim, building on what's been said already, how about something like:
* Keep `ActorSetPos` (I still think there's value in having these lines emitted separately.)
* When in combat, pipe `ActorMove` position data for enemies only to `CombatantMemory` and force a `LineCombatant` log write regardless of thresholds/timers (perhaps have it continue to watch enemy position data just in case?  Not sure there's any harm, esp. if the logic is already there).
* Keep existing behavior on `LineCombatant` re: watching/thresholds/timers for non-enemy combatants.  

Thoughts?

**Comment by valarnin** [comment]
My concerns with piping through `CombatantMemory` are:

1. Any delay to refresh other memory data invalidates the point of using this packet data, so we'd be only printing x/y/z/heading anyways, otherwise we're delaying that position data while we poll memory
2. It now becomes impossible to delineate the "memory" position data vs the actual position data. At what point do you trust `CombatantMemory`'s x/y/z/heading position to be 100% accurate, when sometimes it comes from network lines that are correct at that exact timestamp, and sometimes it comes from memory data that may be delayed or mid-interpolation
3. We now have a memory-polling line that relies on network data and opcodes to function
4. Criteria for forcing a position update line will vary compared to criteria for memory lines, so the logic in `CombatantMemory` will become more and more complex
5. `CombatantMemory` is no longer just a representation of the combatant's state in game memory. Instead it's sometimes memory-only, and sometimes fed by network data

**Comment by valarnin** [comment]
> I think I'm okay with `ActorMove`'s "predicted position" being different from the "current position" in memory due to interpolation questions, but I have a feeling that triggers would almost always want the "predicted position" to avoid the usual race conditions. Perhaps it could be a separate set of fields, or remain a separate log line. i.e. fundamentally, it's a different property, so it should be separate.

I think you're suggesting that `ActorMove` should remain a separate line in this case then? Sorry, I'm not sure I understand exactly, my brain's a bit fried tonight.

> As for the larger issue of what "LineCombatant" should be for - I don't think a user would look at "LineCombatant" and think "this is specifically for memory data". They'd want the most accurate possible data for the current position of the combatant. For example, in the case of HP, I can't imagine why someone would ever want potentially-stale HP, when there's a faster and more reliable source available, so that would be a prime candidate to wire into LineCombatant.

`LineCombatant` was always intended to be a stopgap measure to cover data that wasn't available already in network log lines. In fact, my first couple of drafts of it locally before I even started working on the PR itself, it only dumped a few fields from the `Combatant` struct that weren't available from network data. I can foresee a future point where we're getting all the relevant data points entirely from network lines, at which point `LineCombatant` may be removed entirely (to help with lag) or may be instead emulated entirely by data fed from network lines.

As it is currently, `LineCombatant` is by far the biggest CPU hog because of the way it has to work by its nature. The only possible way to improve performance of the code there in a noticeable way is to read the data from within the game process, which isn't something I'm willing to do.

(the second biggest CPU hog is ironically the actual DPS meter code, it's something I need to look at eventually, something to do with the way the data has to be formatted IIRC)

**Comment by wexxlee** [comment]
Still thinking about the discussion, but wanted to share some data in the meantime:

| Instance         | Line Type   | Total Lines (Size) | Enemy Lines (Size) |
| ---------------- | ----------- | ------------------ | ------------------ |
| Lunar Subterrane | All         | 79790 (10.36M)     |                    |
|                  | ActorSetPos | 167 (17.1K)        | 162 (16.6K)        |
|                  | ActorMove   | 37769 (3.94M)      | 6192 (661K)        |
| DSR (11m P6 Wipe)| All         | 83707 (12.72M)     |                    |
|                  | ActorSetPos | 888 (89.3K)        | 887 (89.2K)        |
|                  | ActorMove   | 21611 (2.19M)      | 285 (29.7K)        |

**Comment by valarnin** [comment]
Some more super quick info. I implemented a "only emit player `ActorMove` lines every 500ms" logic, except instead of ignoring the packet I replaced the content with `SUPPRESSED` and emitted the line anyways, to get actual numbers for what something like that would look like:

```
$ wc -l syrcus\ tower.log
181404 syrcus tower.log
$ cut -d'|' -f1 syrcus\ tower.log | sort -n | uniq -c
  14554 00
      1 01
      1 02
    398 03
    338 04
     13 11
      2 12
   1838 20
   7634 21
   3932 22
    138 23
   2569 24
     94 25
   6415 26
     39 27
   2877 30
    142 31
     18 33
    145 34
      7 35
    214 36
  11157 37
  10091 38
   5852 39
      7 40
     46 41
      8 42
     28 260
  14537 261
   1838 263
   8362 264
      1 265
      8 266
     12 267
  87789 270
    300 271

$ grep -Pv '\|SUPPRESSED\|' syrcus\ tower.log | wc -l                            
115966
$ grep -Pv '\|SUPPRESSED\|' syrcus\ tower.log | cut -d'|' -f1 | sort -n | uniq -c
  14554 00
      1 01
      1 02
    398 03
    338 04
     13 11
      2 12
   1838 20
   7634 21
   3932 22
    138 23
   2569 24
     94 25
   6415 26
     39 27
   2877 30
    142 31
     18 33
    145 34
      7 35
    214 36
  11157 37
  10091 38
   5852 39
      7 40
     46 41
      8 42
     28 260
  14537 261
   1838 263
   8362 264
      1 265
      8 266
     12 267
  22350 270
    300 271
```

Every `ActorMove` results in 48% of log lines being `ActorMove`.
With suppress logic, `ActorMove` is 19% of log lines.

```
$ grep -P '^270\|' syrcus\ tower.log | cut -d'|' -f3 | cut -d'0' -f1 | sort | uniq -c 
  81954 1
   5835 4

$ grep -P '^270\|' syrcus\ tower.log | grep -v SUPPRESSED | cut -d'|' -f3 | cut -d'0' -f1 | sort | uniq -c
  16515 1
   5835 4
```

Before suppression is applied, 92% of `ActorMove` lines are from players.
After suppression is applied, 73% of `ActorMove` lines are from players.

---

I think that this is workable, but the threshold for suppression needs to be higher. Maybe 1 second or 5 seconds makes more sense.

Realistically, we don't need realtime player position info for anything, so we could in fact just not emit any player `ActorMove` lines at all.

```
$ grep -Pv '^270\|[^|]+\|1' syrcus\ tower.log | wc -l                                                     
99451
$ grep -Pv '^270\|[^|]+\|1' syrcus\ tower.log | grep -P '^270' | wc -l
5835
```

Which would result in 5% of lines being `ActorMove` lines.

This could be further trimmed by not emitting lines for player pets, but that filtering is a bit more complex (both C#-side and trying to figure that out after the fact from log files).

**Comment by xpdota** [comment]
> Realistically, we don't need realtime player position info for anything, so we could in fact just not emit any player ActorMove lines at all.

I think this is reasonable. LineCombatant already covers normal player movements well enough.

> This could be further trimmed by not emitting lines for player pets, but that filtering is a bit more complex (both C#-side and trying to figure that out after the fact from log files).

I feel like it could be done "well enough" by just filtering out specific bNpcIds. Not like there's that many different player summons.

**Comment by wexxlee** [comment]
> Which would result in 5% of lines being `ActorMove` lines.

To be fair, there's probably some variability in number that depending on content type.  In Lunar Subterrane, enemy `ActorMove` lines were closer to 13%, but that's not surprising considering the packs of mobs that get dragged around.  And on the other end of the spectrum, 8-person content with a single targetable enemy combatant that largely stays put would probably come in well below 5%.  (In an 11m DSR pull, it was <0.5%).  In any event, those numbers aren't crazy excessive imho.

> Realistically, we don't need realtime player position info for anything, so we could in fact just not emit any player `ActorMove` lines at all.

This is where I'm leaning.  I don't think there's a strong use case for player `ActorMove` lines, especially since (a) `LineCombatant` data is available, (b) player position is, even if slightly stale, tracked on plenty of other lines, and (c) people have found adequate ways to use existing player data for what they need (specifically, I'm thinking of fflogs' replay function).  

> This could be further trimmed by not emitting lines for player pets, but that filtering is a bit more complex (both C#-side and trying to figure that out after the fact from log files).

Eh, I don't think that's worth the effort.  It's going to be a very small number of log lines, and I think filtering based on actorId (1.{7} vs. 4.{7}, or however you'd choose to implement it) is going to be faster and cleaner than trying to identify particular actors to exclude (not to mention potentially having to change that logic when SqEx invariably changes/introduces new pets).

I still have a sort of ephemeral concern about the fact that there's a large overlap between `ActorMove` and `LineCombatant` -- but your reasons for not piping seem sound to me.  Understanding that `LineCombatant` was intended as a stop-gap, there is already some reliance on it now, so I'm not sure I have an opinion yet whether, down the road, it would make more sense to deprecate & eliminate the line or emulate the data.  But it's not like its unprecedented to have overlapping data on different log lines currently, so given this volume data, I'm comfortable with emitting `ActorMove` only for NPC combatants and keeping `ActorSetPos` as is, fwiw.

**Comment by valarnin** [comment]
> I feel like it could be done "well enough" by just filtering out specific bNpcIds. Not like there's that many different player summons.

We don't have that data on the network packet, it would have to be collected elsewhere, and the only consistent way to do so would be memory polling via `getCombatants`. Then we'd have to track zone init lines to reset the cached values or else we'd end up filtering out the wrong combatants after a zone change or wipe.

It's not a case of "how do we identify if it's a summon", it's more a case of "how do we get and track the data points that allow us to identify if it's a summon, without excluding correct actors".

File: OverlayPlugin.Core/NetworkProcessors/LineActorSetPos.cs
Comment (wexxlee) [inline OverlayPlugin.Core/NetworkProcessors/LineActorSetPos.cs:21]: Is there a difference between `rotation` and `heading`?  I see `heading` in the `ActorMove` struct and that's how I've seen it referred to, so not sure if `rotation` has a different meaning (if so, an explanatory comment would be great).

File: OverlayPlugin.Core/NetworkProcessors/LineActorSetPos.cs
Comment (wexxlee) [inline OverlayPlugin.Core/NetworkProcessors/LineActorSetPos.cs:46]: Should these be flipped in `ActorMove` as well?

**Review by wexxlee — APPROVED** [review]
This looks good to me.  Thanks for adding these.  A couple small questions/clarifications, but I'll defer to you if changes are warranted.

File: OverlayPlugin.Core/NetworkProcessors/LineActorSetPos.cs
Comment (valarnin) [reply OverlayPlugin.Core/NetworkProcessors/LineActorSetPos.cs:21]: Traditionally:
- `rotation` would be a value from `-π` to `+π`
- `heading` would be a 2D or 3D vector

But the language is mixed elsewhere in the project.

- `FFXIV_ACT_Plugin` uses `rotation`.
- `FFXIVClientStructs` uses `rotation`.
- `OverlayPlugin`'s combatant-related classes and structures use `heading`.
- `OverlayPlugin`'s `LineActorCastExtra` and `LineAbilityExtra` use `rotation`.

I think it would make sense for everything new going forward to use the correct terminology of `rotation`.

**Pull request merged**
